grumphp.yml.dist

```
@@ -9,6 +9,7 @@ parameters:
     yamllint: ~
     composer: ~
     jsonlint: ~
+    twigcs: ~
```

Drupal core doesn't conform to the twigcs rules and would throw a lot of errors. We could exclude core and contrib to work around it. The other option, which I'm not completely in favour of, would be to create a custom ruleset that matches the way Drupal core formats twig files.

---

We should add the `path` option set to `web/themes/custom`.

---

@hussainweb That would not cover templates in custom/modules.

---

@skippednote, sure, but unless we have a better solution, this is valuable at least. I feel we could go ahead with this now as default and figure out if grumphp can run tasks repeatedly.

---

@hussainweb My suggestion is that we set the path to `web` or `.` and ignore `contrib/*` and `core` .

---

It looks like it's already possible. https://github.com/phpro/grumphp/blob/master/doc/tasks.md#run-the-same-task-twice-with-different-configuration

---

why run twice when we can handle it in a single run?
```suggestion
    twigcs: 
      path: '.'
      exclude:
        - core
        - contrib/*
```

---

> My suggestion is that we set the path to web
@skippednote, the problem is that will include core and contribs. How would you ignore them?

I feel that running the task twice, once with modules and then with themes might work better.

---

Ah, I see it has an option to exclude. That works too. We should list every possible `contrib` path, no?

---

>@skippednote, the problem is that will include core and contribs. How would you ignore them?
Check the example above, we can use `exclude` to ignore paths.

---

```suggestion
    twigcs:
      path: 'web'
      exclude:
        - core
        - modules/contrib
        - themes/contrib
        - profiles/contrib
```

---

The paths are supposed to be relative from the [documentation](https://github.com/phpro/grumphp/blob/master/doc/tasks/twigcs.md).

> This option will specify which relative subfolders or files will be exclude to this task.